src/course-outline/card-header/TitleButton.jsx

```diff
@@ -0,0 +1,76 @@
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  Button,
+  Icon,
+  OverlayTrigger,
+  Tooltip,
+  Truncate,
+} from '@edx/paragon';
+import {
+  ArrowDropDown as ArrowDownIcon,
+  ArrowDropUp as ArrowUpIcon,
+} from '@edx/paragon/icons';
+import classNames from 'classnames';
+
+import { ITEM_BADGE_STATUS } from '../constants';
+import { getItemStatusBadgeContent } from '../utils';
+import messages from './messages';
+
+const TitleButton = ({
+  title,
+  status,
+  isExpanded,
+  onTitleClick,
+  namePrefix,
+}) => {
+  const intl = useIntl();
+  const { badgeTitle, badgeIcon } = getItemStatusBadgeContent(status, messages, intl);
+  const titleTooltipMessage = intl.formatMessage(messages.expandTooltip);
+
+  return (
+    <OverlayTrigger
+      placement="bottom-start"
+      overlay={(
+        <Tooltip
+          id={titleTooltipMessage}
+          className="item-card-header-tooltip"
+        >
+          {titleTooltipMessage}
+        </Tooltip>
+      )}
+    >
+      <Button
+        iconBefore={isExpanded ? ArrowUpIcon : ArrowDownIcon}
+        variant="tertiary"
+        data-testid={`${namePrefix}-card-header__expanded-btn`}
+        className="item-card-header__expanded-btn"
+        onClick={onTitleClick}
+      >
+        <Truncate lines={1} className={`${namePrefix}-card-title mb-0`}>{title}</Truncate>
+        {badgeTitle && (
+          <div className="item-card-header__badge-status" data-testid={`${namePrefix}-card-header__badge-status`}>
+            {badgeIcon && (
+              <Icon
+                src={badgeIcon}
+                size="sm"
+                className={classNames({ 'text-success-500': status === ITEM_BADGE_STATUS.live })}
+              />
+            )}
+            <span className="small">{badgeTitle}</span>
+          </div>
+        )}
```

Comment:
@navinkarkera The number of lines of code seems little, but there are many details in them. I feel like it is enough to extract it into a base component `Title` (or `BaseTitle`), to avoid duplication and having to mirror the changes in two places.